Neos.Neos/Classes/Fusion/AbstractMenuItemsImplementation.php

```diff
@@ -150,18 +152,19 @@ protected function isNodeHidden(Node $node)
     protected function getCurrentNodeRootline(): array
     {
         if ($this->currentNodeRootline === null) {
-            /** @todo replace this */
-            /*
-            $nodeRootline = $this->currentNode->getContext()->getNodesOnPath(
-                $this->runtime->getCurrentContext()['site']->getPath(),
-                $this->currentNode->getPath()
-            );
-            $this->currentNodeRootline = [];
-
-            foreach ($nodeRootline as $rootlineElement) {
-                $this->currentNodeRootline[$this->getNodeLevelInSite($rootlineElement)] = $rootlineElement;
-            }*/
-            $this->currentNodeRootline = [];
+            $rootline = [$this->currentNode];
+            foreach (
+                $this->contentRepositoryRegistry->subgraphForNode($this->currentNode)
+                    ->findAncestorNodes(
+                        $this->currentNode->nodeAggregateId,
+                        FindAncestorNodesFilter::create()
+                    ) as $ancestorNode
+            ) {
+                if (!$ancestorNode->classification->isRoot()) {
+                    $rootline[] = $ancestorNode;
```

Comment:
previously we included the depth here as index ...

Reply:
should be fixed now